sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -3982,6 +3982,14 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
       messageParameters = Map("provider" -> name))
   }
 
+  def externalDataSourceException(cause: Option[Throwable]): Throwable = {
+    new AnalysisException(
+      errorClass = "DATA_SOURCE_EXTERNAL_ERROR",
+      messageParameters = Map("error" -> cause.getOrElse("unknown").getClass.toString),
```

Comment:
You already added `cause` as a parameter to `AnalysisException`. Let's don't 1. duplicate error message; 2. avoid embedding a random uncontrolled text to error messages.

Reply:
removed